In [20]:
import pandas as pd
import plotnine as p9
import numpy as np
import os

### Read Files

In [16]:
df_act_leave = pd.read_excel('../../SharePoint/HR Analytics - Documents/Capstone Data/V2 data 5.1.2020/Active + Leave (Weekly) - 2020_04_28.xlsx')

## Active + Leaves (weekly)

In [23]:
df_act_leave.head()

Employee Code     Original Hire Date Recent Hire Date (if applicable)  \
0           A12  Monday, March 6, 2006            Monday, March 6, 2006   
1           A12  Monday, March 6, 2006            Monday, March 6, 2006   
2           A12  Monday, March 6, 2006            Monday, March 6, 2006   
3           A12  Monday, March 6, 2006            Monday, March 6, 2006   
4           A12  Monday, March 6, 2006            Monday, March 6, 2006   

  Gender      Job Title Position Status      Worker Category  Birth Year  \
0   Male  SALES MANAGER          Active  Permanent Full Time        1981   
1   Male  SALES MANAGER          Active  Permanent Full Time        1981   
2   Male  SALES MANAGER          Active  Permanent Full Time        1981   
3   Male  SALES MANAGER          Active  Permanent Full Time        1981   
4   Male  SALES MANAGER          Active  Permanent Full Time        1981   

  Report Date (week ending)        Generation  Tenure Tenure Group  QTY  
0   Friday, January 5, 2018  Gen X: 1965-1982    4323  10-15 Years    1  
1  Friday, January 12, 2018  Gen X: 1965-1982    4330  10-15 Years    1  
2  Friday, January 19, 2018  Gen X: 1965-1982    4337  10-15 Years    1  
3  Friday, January 26, 2018  Gen X: 1965-1982    4344  10-15 Years    1  
4  Friday, February 2, 2018  Gen X: 1965-1982    4351  10-15 Years    1

In [33]:
df_act_leave.tail()

Employee Code     Original Hire Date Recent Hire Date (if applicable)  \
204138           ZW1  Monday, June 19, 2017            Monday, June 19, 2017   
204139           ZW1  Monday, June 19, 2017            Monday, June 19, 2017   
204140           ZW1  Monday, June 19, 2017            Monday, June 19, 2017   
204141           ZW1  Monday, June 19, 2017            Monday, June 19, 2017   
204142           ZW1  Monday, June 19, 2017            Monday, June 19, 2017   

        Gender      Job Title Position Status      Worker Category  \
204138  Female  SALES MANAGER          Active  Permanent Full Time   
204139  Female  SALES MANAGER          Active  Permanent Full Time   
204140  Female  SALES MANAGER          Active  Permanent Full Time   
204141  Female  SALES MANAGER          Active  Permanent Full Time   
204142  Female  SALES MANAGER          Active  Permanent Full Time   

        Birth Year Report Date (week ending)            Generation  Tenure  \
204138        1985    Friday, March 27, 2020  Millenial: 1983-1999    1012   
204139        1985     Friday, April 3, 2020  Millenial: 1983-1999    1019   
204140        1985    Friday, April 10, 2020  Millenial: 1983-1999    1026   
204141        1985    Friday, April 17, 2020  Millenial: 1983-1999    1033   
204142        1985    Friday, April 24, 2020  Millenial: 1983-1999    1040   

       Tenure Group  QTY  
204138    2-3 Years    1  
204139    2-3 Years    1  
204140    2-3 Years    1  
204141    2-3 Years    1  
204142    2-3 Years    1

In [36]:
#check shape
df_act_leave.shape

(204143, 13)

In [31]:
#unqiue employees codes  

emp_code_uni = len(df_act_leave["Employee Code"].unique().tolist())
print(f"Unique Employee Codes: {emp_code_uni}")

Unique Employee Codes: 3435


In [35]:
# types of gender 

df_act_leave["Gender"].unique()

array(['Male', 'Female', 'Not Specified'], dtype=object)

In [39]:
#position types 
df_act_leave["Job Title"].unique()

array(['SALES MANAGER', 'DISTRICT TEAM LEADER', 'SALES ASSOCIATE',
       'ASSISTANT MANAGER', 'MANAGER IN TRAINING', 'TEMPORARY MANAGER',
       'MULTIPLE STORE MANAGER', 'ACTING MANAGER',
       'APPLE SERVICE TECHNICIAN', 'APPLE REPAIR TECHNICIAN'],
      dtype=object)

In [40]:
df_act_leave["Position Status"].unique()

array(['Active', 'Leave'], dtype=object)

In [41]:
df_act_leave["Worker Category"].unique()

array(['Permanent Full Time', 'Permanent Part Time',
       'Temporary Part Time', 'Temporary Full Time'], dtype=object)

In [44]:
df_act_leave["Generation"].unique()

array(['Gen X: 1965-1982', 'Millenial: 1983-1999',
       'Baby Boomer: 1945-1964', 'Gen Z: 2000-Present'], dtype=object)

In [45]:
df_act_leave["Tenure Group"].unique()

array(['10-15 Years', '15+ Years', '5-10 Years', '3-5 Years', '2-3 Years',
       'Under 1 Year', '1-2 Years'], dtype=object)

In [46]:
df_act_leave["QTY"].unique()

array([1], dtype=int64)

### Summary Data Table

Number of rows and columns
    - Rows: 204143
    - Columns: 13

Columns:
- Employee Code: Three digit code linking the table unique employee codes 
    - unqique employees 
- Original hire date: day in a wday, month, day, year 
- Recent Hire Date (if applicable): day in a wday, month, day, year usually matches the Original Hire Date
- Gender: employees' geneder values: [male, female, non specified]
- Job Titles: employees' job title some values: ['SALES MANAGER', 'DISTRICT TEAM LEADER', 'SALES ASSOCIATE', 'ASSISTANT MANAGER', 'MANAGER IN TRAINING']
- Position Status: If they are `Active` in a given week or on `Leave`
- Worker Category: employeement status of employee values= ['Permanent Full Time', 'Permanent Part Time', 'Temporary Part Time', 'Temporary Full Time']
- Birth year: employees' birth year
- Report Date: date of the report the entry was pulled from
- Generation: employees' generation values = ['Gen X: 1965-1982', 'Millenial: 1983-1999', 'Baby Boomer: 1945-1964', 'Gen Z: 2000-Present']
- Tenure: employees' code tenure 
- Tenure Group: employees years with the company some values ['10-15 Years', '15+ Years', '5-10 Years']
- QTY it always has a value of `1`

In [47]:
df_act_leave.groupby()

Employee Code                       object
Original Hire Date                  object
Recent Hire Date (if applicable)    object
Gender                              object
Job Title                           object
Position Status                     object
Worker Category                     object
Birth Year                           int64
Report Date (week ending)           object
Generation                          object
Tenure                               int64
Tenure Group                        object
QTY                                  int64
dtype: object


**No more graphs could made**